test/apps/iodemo/io_demo.cc

```diff
@@ -442,11 +449,12 @@ class P2pDemoCommon : public UcxContext {
         long*                             _op_counter;
         size_t                            _counter;
         BufferIov*                        _iov;
+        IoMessage*                        _io_msg;
         MemoryPool<SendCompleteCallback>& _pool;
     };
 
     P2pDemoCommon(const options_t& test_opts) :
-        UcxContext(test_opts.iomsg_size, test_opts.connect_timeout),
+        UcxContext(test_opts.iomsg_size, test_opts.connect_timeout, test_opts.use_am),
```

Comment:
enum ID values could stay the same for TAG io messages but with your approach the protocol will be broken if you mean backward compatibility in case when client and server have different versions.